io/io/src/RRawFileUnix.cxx

```diff
@@ -115,8 +115,60 @@ void ROOT::Internal::RRawFileUnix::ReadVImpl(RIOVec *ioVec, unsigned int nReq)
 {
 #ifdef R__HAS_URING
    if (RIoUring::IsAvailable()) {
-      // todo(max) actually use the ring
-      RRawFile::ReadVImpl(ioVec, nReq);
+      RIoUring ring(nReq);
+      struct io_uring *p_ring = ring.raw();
```

Comment:
I think we want to use the uring in several places eventually.  So we should build a light API in the `RIoUring` class that lets us drop in a vector (span) of requests and wait for completion.  The request type would probably look similar to the `RRawFile::RIOVec` struct, perhaps with the addition of a file descriptor (and later more if we also support writing).

Reply:
Thank you Jakob, I made a new struct `RReadEvent` for this purpose: 
```cpp
struct RReadEvent { 
   RRawFile::RIOVec fIoVec; 
   int fFileDes; 
}; 
```
and added a new `SubmitReadsAndWait(std::vector<RReadEvent>& readEvents)` method in 464f7f5 

It might be the case that `RReadEvent` can do double duty as a `WriteEvent` too, `RIOVec` is pretty versatile.